test/contrib/emscripten/conftest.py

```diff
@@ -90,44 +127,97 @@ def _install_coverage(self: Any) -> None:
 
 
 class ServerRunnerInfo:
-    def __init__(self, host: str, port: int, selenium: Any) -> None:
+    def __init__(self, host: str, port: int, selenium: Any, dist_dir: Path) -> None:
         self.host = host
         self.port = port
         self.selenium = selenium
+        self.dist_dir = dist_dir
 
     def run_webworker(self, code: str) -> Any:
         if isinstance(code, str) and code.startswith("\n"):
             # we have a multiline string, fix indentation
             code = textwrap.dedent(code)
-            # add coverage collection to this code
-            code = (
-                textwrap.dedent(
-                    """
-            import coverage
-            _coverage= coverage.Coverage(source_pkgs=['urllib3'])
-            _coverage.start()
-            """
-                )
-                + code
-            )
-            code += textwrap.dedent(
+        # replace imports of urllib3 in the code
+        # so that pyodide doesn't find them and automatically
+        # load the urllib3 module from its distribution
+        # n.b. we can't override this import otherwise because
+        # we can't run custom js in the pyodide worker test template
+        # before the python code is launched
```

Comment:
It might be easier to replace `webworker.js`. It's only 26 lines and quite simple. But this seems fine if a bit regrettable.

Reply:
yeah, I did that now.